### ISSUE TITLE
Issue while replicating the finetuning results

### ISSUE THREAD

**Issue by gauthier-th** [sunlicai/MAE-DFER#21]
Hi, thanks for the great work!

I am currently trying to replicate the results of the finetuning on the DFEW dataset, we the pretrained model you provided.

With the DFEW dataset I don't manage to reach the reported results, ending up having lower results by a small margin.

|  Fold    | UAR        | WR       |
| :------: | :--------: | :------: |
|  1       | 62.13      | 74.58    |
|  2       | 60.98      | 71.89    |
|  3       | 63.75      | 74.23    |
|  4       | 62.43      | 73.66    |
|  5       | 63.75      | 75.68    |
|  My total   | 62.61      | 74.01    |
|  Reported total   | 63.41    | 74.43 |
| **Difference** | **0.80** | **0.42** |

I tried to modify the number of GPUs, the batch size and the number of epochs, without any significant impact on this margin.

Thank you in advance for any tips or solutions you can provide.

**Comment by sunlicai** [comment]
Hi,

Thank you for your interest in our work!

May I ask if you used time-normalized clips (16 frames per sample), or the original variable-length clips (our defaults)?

You can tune other parameters, e.g., learning rate, mixup, and smoothing. You can also check whether your key packages in [environment.yml](https://github.com/sunlicai/MAE-DFER/blob/master/environment.yml) (torch, einops, timm, numpy, etc.) and hardware (Nvidia V100 or 3090) match our defaults.

**Comment by gauthier-th** [comment]
Hi, thanks for your prompt reply.

I also used the variable-length clips. I will definitely try to tune other parameters to see if I can get an improvement - just wanted to know if I missed a parameter to change in the finetuning script.

However I did update the packages to a newer version, as well as the Python version (3.11), with another hardware (Nvidia H100).
```
decord==0.6.0
deepspeed==0.17.1
einops==0.8.1
opencv-python==4.11.0.86
openpyxl==3.1.5
pandas==2.3.0
scikit-image==0.25.2
scikit-learn==1.7.0
tensorboardX==2.6.4
timm==0.4.12
torch==2.7.1
torchaudio==2.7.1
torchvision==0.22.1
```
I'll try to make it run with the default packages in Python 3.8.

**Comment by gauthier-th** [comment]
I ran the experiment again with the exact same conda environment provided in the repo, and with 3 Nvidia V100 GPUs, but I'm still having results lower than expected:

UAR: 62.908% (-0.5%), WAR: 74.14% (-0.3%)

I ran it on the original variable-length clips, with the same parameters as the ones in the example script.

**Comment by sunlicai** [comment]
As the gap is quite small, I believe that it is acceptable considering the inherent randomness. You can try to tune parameters on the relatively low-performance folds to get better results.
